niceman/distributions/conda.py

```diff
@@ -243,17 +242,32 @@ def _get_conda_info(self, conda_path):
                         conda_path, exc_str(exc))
         return details
 
-    def _get_conda_path(self, path):
-        return self._path_root(path)
+    def _get_conda_env_path(self, path):
+        return self._env_path_root(path)
 
-    def _is_conda_directory(self, path):
+    def _is_conda_env_directory(self, path):
         try:
             _, _ = self._session.execute_command(
-                'ls -ld %s/bin/conda %s/conda-meta'
-                % (path, path)
+                'ls -ld %s/conda-meta'
+                % path
             )
         except Exception as exc:
-            lgr.debug("Did not detect conda at the path %s: %s", path,
+            lgr.debug("Did not detect conda env at the path %s: %s", path,
+                      exc_str(exc))
+            return False
+        return True
+
+    def _get_conda_root_path(self, path):
+        return self._root_path_root(path)
+
+    def _is_conda_root_directory(self, path):
+        try:
+            _, _ = self._session.execute_command(
+                'ls -ld %s/bin %s/envs %s/conda-meta'
+                % (path, path, path)
+            )
```

Comment:
I like using session.exists because it would be faster for local sessions (though we don't have ShellSession.exists() local function defined (I'll do that now).